src/libDirectoryService/DirectoryService.cpp

```diff
@@ -1128,6 +939,120 @@ bool DirectoryService::ProcessAllPoWConnResponse(
         m_hasAllPoWconns = true;
     }
     cv_allPowConns.notify_all();
+
+    return true;
+}
+
+// Current this is only used by ds. But ideally, 20 ds nodes should
+bool DirectoryService::ProcessAllPoW2Request(
+    const vector<unsigned char>& message, unsigned int offset, const Peer& from)
+{
+    LOG_MARKER();
+    LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "I am sending AllPow2 to requester");
+
+    lock(m_mutexAllPOW1, m_mutexAllPOW2, m_mutexAllPoWConns);
```

Comment:
Hmm wasn't this process of populating m_allPoW2s already done in DirectoryService::ProcessDSBlockConsensusWhenDone()?  Or is this particular DS node not going to perform that code?